documentation/modules/ROOT/pages/03-demo.adoc

```diff
@@ -7,28 +7,879 @@
+            .send()
+            .await
+            .map_err(error::ErrorInternalServerError)?;
+
+        match response.status().is_success() {
+            true => Ok(()),
+            false => {
+                log::error!("failed to send event: {:#?}", response);
+                Err(error::ErrorInternalServerError(format!(
+                    "failed to send event: {}",
+                    response.status()
+                )))
+            }
+        }
+    }
+}
+----
+
+<1> The client uses _POST_ method, to send the _JSON_ representation of the event to the sink.
+The _sink_ is the URL of the target, in this case the url of the _Event Mesh_.
+
+==== Event Mesh
+
+In this section, we'll use the _Event Mesh_ setup to communication between the extracted Drivers module and the different parts of the application.
+
+Here's the configuration of the _Event Mesh_'s central component, the _Broker_, which will be used in this example.
+The _Broker_ here is the _Knative_ component, and will be deployed in the _Kubernetes_ cluster.
+
+[source,yaml]
+----
+apiVersion: eventing.knative.dev/v1
+kind: Broker
+metadata:
+  name: default
+  namespace: demo
+spec:
+  delivery:
+    backoffDelay: PT0.2S # <1>
+    backoffPolicy: exponential # <2>
+    retry: 10 # <3>
+----
+
+<1> The `+backoffDelay+` is the delay between retries, and us use `+200ms+` initially.
+<2> The `+backoffPolicy+` is set to `+exponential+`, which means that the delay will be doubled each time.
+<3> The `+retry+` is the number of times we retry before giving up.
+
+[IMPORTANT]
+====
+The policy is `+exponential+`, and the `+retry+` is 10, which means that after approximately 6 min and 50 sec the event will be dropped.
+====
+
+[NOTE]
+====
+A `+deadLetterSink+` option could be configured for the _Broker_ to send the events that failed to be delivered in time to a back-up location.
+Events captured in a back-up location can be re-transmitted into the _Event Mesh_ later.
+====
+
+==== Legacy application changes
+
+The last part of the refactoring will be the changes needed in our legacy Java application.
+We need to remove the _Drivers_ logic and send events to the _Event Mesh_ instead.
+We also need to accept new events coming from the _Event Mesh_, as the calculated fee will be transmitted as such.
+
+Here's the refactored `+completeTransit+` method:
+[source,java]
+----
+public void completeTransit(UUID requestUUID, AddressDTO destinationAddress) {
+    // ...
+    Money finalPrice = completeTransitService.completeTransit(driverId, requestUUID, from, to);
+    // ...
+    driverFeeService.calculateDriverFee(requestUUID, finalPrice, driverId); // <1>
+    // ...
+}
+
+@EventListener // <2>
+public void driverFeeCalculated(DriverFee driverFee) { // <3>
+    Objects.requireNonNull(driverFee.ctx.getSubject());
+    UUID id = UUID.fromString(driverFee.ctx.getSubject());
+    transitDetailsFacade.driverFeeCalculated(id, driverFee.data.fee);
```

Comment:
facade :-)

Reply:
I'm not the original author of [the Ride application](https://github.com/cardil/legacyfighter-cabs-java/tree/bf576d61b679fd7e60b8668cb3bff7333e54fb7f) :-P I just took it, and refactored. I thought it represents quite well what's there in the wild.

![image](https://github.com/user-attachments/assets/1c44cb83-094d-4ea3-99de-317605d984e1)